hydrolib/core/dflowfm/ext/models.py

```diff
@@ -181,6 +188,67 @@ def validate_location_type(cls, v: str) -> str:
         return v
 
 
+class SourceSink(INIBasedModel):
+    """
+    A `[SourceSink]` block for use inside an external forcings file,
+    i.e., a [ExtModel][hydrolib.core.dflowfm.ext.models.SourceSink].
+
+    All lowercased attributes match with the lateral input as described in
+    [UM Sec.C.5.2.4](https://content.oss.deltares.nl/delft3dfm1d2d/D-Flow_FM_User_Manual_1D2D.pdf#subsection.C.5.2.4).
+    """
+
+    _header: Literal["Lateral"] = "SourceSink"
+    id: str = Field(alias="id")
+    name: str = Field("", alias="name")
+    locationfile: DiskOnlyFileModel = Field(
+        default_factory=lambda: DiskOnlyFileModel(None), alias="locationFile"
+    )
+
+    numcoordinates: Optional[int] = Field(alias="numCoordinates")
+    xcoordinates: Optional[List[float]] = Field(alias="xCoordinates")
+    ycoordinates: Optional[List[float]] = Field(alias="yCoordinates")
+
+    zsource: Optional[Union[float, List[float]]] = Field(alias="zSource")
+    zsink: Optional[Union[float, List[float]]] = Field(alias="zSink")
+    discharge: Union[float, List[float]] = Field(alias="discharge")
+    area: Optional[float] = Field(alias="Area")
+
+    salinitydelta: Optional[Union[List[float], float]] = Field(alias="SalinityDelta")
+    temperaturedelta: Optional[Union[List[float], float]] = Field(
```

Comment:
Also use type `ForcingData` instead of the custom Union,

Reply:
The `ForcingData` class did not have a `List[float]` so now I used it after i added `List[float]` to it

Reply:
See my new remark over there for forcingData.

Reply:
this is now back to Union[float, List[floats]] and will be changes to use `forcingData` in a separate merge